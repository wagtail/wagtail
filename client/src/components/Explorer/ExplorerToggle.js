import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';

import * as actions from './actions';

import Button from '../../components/Button/Button';

/**
 * A Button which toggles the explorer.
 */
const ExplorerToggle = ({ children, onToggle }) => (
  <Button
    className="submenu-trigger"
    icon="folder-open-inverse"
    dialogTrigger={true}
    onClick={onToggle}
  >
    {children}
  </Button>
);

ExplorerToggle.propTypes = {
  onToggle: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
};

const mapStateToProps = () => ({});

const mapDispatchToProps = (dispatch) => ({
  onToggle: (page) => dispatch(actions.toggleExplorer(page)),
});

const mergeProps = (stateProps, dispatchProps, ownProps) => ({
  children: ownProps.children,
  onToggle: dispatchProps.onToggle.bind(null, ownProps.startPage),
});

export default connect(mapStateToProps, mapDispatchToProps, mergeProps)(ExplorerToggle);
