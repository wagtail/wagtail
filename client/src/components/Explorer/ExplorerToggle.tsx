 

import React from 'react';
import { connect } from 'react-redux';

import * as actions from './actions';

import Button from '../../components/Button/Button';
import Icon from '../../components/Icon/Icon';

interface ExplorerToggleProps {
  onToggle(): void;
  children: React.ReactNode;
}

/**
 * A Button which toggles the explorer.
 */
const ExplorerToggle: React.FunctionComponent<ExplorerToggleProps> = ({ children, onToggle }) => (
  <Button
    dialogTrigger={true}
    onClick={onToggle}
  >
    <Icon name="folder-open-inverse" className="icon--menuitem" />
    {children}
    <Icon name="arrow-right" className="icon--submenu-trigger" />
  </Button>
);

const mapStateToProps = () => ({});

const mapDispatchToProps = (dispatch) => ({
  onToggle: (page) => dispatch(actions.toggleExplorer(page)),
});

const mergeProps = (_stateProps, dispatchProps, ownProps) => ({
  children: ownProps.children,
  onToggle: dispatchProps.onToggle.bind(null, ownProps.startPage),
});

export default connect(mapStateToProps, mapDispatchToProps, mergeProps)(ExplorerToggle);
