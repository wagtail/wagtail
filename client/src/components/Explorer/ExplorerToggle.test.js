import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';

import ExplorerToggle from './ExplorerToggle';

const store = configureMockStore()({});

describe('ExplorerToggle', () => {
  it('exists', () => {
    expect(ExplorerToggle).toBeDefined();
  });

  it('basic', () => {
    expect(shallow((
      <ExplorerToggle store={store}>
        <span>
          To infinity and beyond!
        </span>
      </ExplorerToggle>
    )).find('ExplorerToggle').dive()).toMatchSnapshot();
  });

  describe('actions', () => {
    let wrapper;

    beforeEach(() => {
      store.dispatch = jest.fn();
      wrapper = shallow(<ExplorerToggle store={store}>Test</ExplorerToggle>);
    });

    it('onToggle', () => {
      wrapper.dive().prop('onToggle')();
      expect(store.dispatch).toHaveBeenCalled();
    });
  });
});
