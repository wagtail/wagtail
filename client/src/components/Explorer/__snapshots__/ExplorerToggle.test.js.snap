// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExplorerToggle basic 1`] = `
<Button
  accessibleLabel={null}
  className="submenu-trigger"
  dialogTrigger={true}
  href="#"
  icon="folder-open-inverse"
  isLoading={false}
  onClick={[Function]}
  preventDefault={true}
  target={null}
>
  <span>
    To infinity and beyond!
  </span>
</Button>
`;
